api/test-helpers.js

```diff
@@ -124,16 +143,39 @@ class WebSocket {
   send(msg) {
     this.lastRequest = JSON.parse(msg);
     this.requests.push(this.lastRequest);
+    this._autoReply(this.lastRequest['request-id']);
   }
 
-  reply(resp) {
+  _autoReply(requestId) {
+    if (this._queuedResponses.has(requestId)) {
+      this.reply(this._queuedResponses.get(requestId), requestId);
+    }
+  }
+
+  /**
+    Reply to requests from the websocket.
+    @param {Object} resp - The response for the request in a JSON.stringify-able
+      format
+    @param {Number} [requestId] - The request ID that you want to respond to. Optional.
+  */
+  reply(resp, requestId) {
     if (this.lastRequest === null) {
       throw new Error('cannot reply as no requests were received');
     }
-    resp['request-id'] = this.lastRequest['request-id'];
-    this.onmessage({data: JSON.stringify(resp)});
+    resp['request-id'] = requestId || this.lastRequest['request-id'];
+    const response = {data: JSON.stringify(resp)};
+    this.responses.push(response);
```

Comment:
What;s the goal of this.responses? If it is for further inspection, would it be better just adding the unmarshaled resp rather than a string?

Reply:
It's added for debugging purposes....I ran into some issues where the response values I was sending to requests weren't correct and there was no way for me to get a "history" of responses to debug the issue. I wanted to store the actual response but yeah I can see an unmarshalled data value being easier to use...

Reply:
done